exchanges/okx/ws_requests.go

```diff
@@ -383,11 +325,40 @@ func (ok *Okx) SendAuthenticatedWebsocketRequest(ctx context.Context, epl reques
 		return err
 	}
 
-	// Codes 0-2 have responses and can be handled individually by the caller, while codes > 2 are endpoint errors with
-	// messages and no responses, checked here.
-	if intermediary.Code > 2 {
+	switch intermediary.Code {
+	case 0:
+		return nil
+	case 1:
+		return errOperationFailed
+	case 2:
+		return errPartialSuccess
+	default:
 		return getStatusError(intermediary.Code, intermediary.Message)
 	}
+}
 
-	return nil
+func mergeErrorDetails[T interface{ Error() error }](err error, resp []T) error {
```

Comment:
🐒 
* Firstly, this is super cute, nice work 💪 
* _cough_ 🙁  I think maybe it's not necessary
Because you're saying that T is something that implments a new interface which is actuallyt identical to the built in `error` [interface itself](https://go.dev/ref/spec#Errors). Inception time, right... T is an interface that returns an interface that matches itself.
* And I don't think we're really merging errors; More like collecting them. Specifically, we're not putting errors inside each other. And if we were, we'd probably use the idiomatic wrap terminology.
So all told, I'd like, but not insist upon, something that dances like:
```suggestion
func collectErrors(errs error) error {
...
```
And then use common.AppendErrors above and it's nil ignoring to shortcut this.
But then ...

Reply:
Okay, so ... if everything is an error implementing thing... isn't this just the same as calling:
```
common.AppendError(errs, err)
```
over range.

So ... I'm thinking that:
```
common.CollectErrors(err, resps...)
```
could be a thing. Naming clashes with the existing method on ErrorCollector, but this is about turning a slice of errors into a multiError via AppendErrors, so I think it's okay.

</ stream of consciousness >